toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1013,8 +1013,9 @@ func (g *PkgGraph) AddMetaNode(from []*PkgNode, to []*PkgNode) (metaNode *PkgNod
 	return
 }
 
-// AddGoalNode adds a goal node to the graph which links to existing nodes. An empty package list will add an edge to all nodes
-func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer, strict bool) (goalNode *PkgNode, err error) {
+// AddGoalNode adds a goal node to the graph which links to existing nodes. An empty package list will add an edge to all nodes.
+// useSRPMPath will use also include any additional matches based on the SRPM path, not just the package name.
+func (g *PkgGraph) AddGoalNode(goalName string, packages []*pkgjson.PackageVer, strict bool, useSRPMPath bool) (goalNode *PkgNode, err error) {
```

Comment:
I was actually debating using this feature in the scheduler as well, since it makes it more 'accurate' to life. When we request a single sub-package `foo-sub1.rpm` from the scheduler what we will actually get is `foo.rpm`, `foo-sub1.rpm`, and `foo-sub2.rpm`. We generate all of them since they all originate from a single SRPM.

As for the fetcher, that is where this behavior will be used. There is a fair bit of logic involved in optimizing the graph and I reuse the scheduler library in the new fetcher changes. (Hence wanting to just enhance the goal node code rather than re-writing the library).

Reply:
I think I can get my changes to work without this. Marking as draft for now.